### PR TITLE
custom red metric tags

### PR DESCRIPTION
**Pull request by djia-vm-wf**
issues #14 #16

### CODE REVIEW

File: reporter/reporter.go [hunk 1]
```diff
@@ -236,6 +245,17 @@ func (t *reporter) reportDerivedMetrics(span tracer.RawSpan) {
 	replaceTag(tags, "application", appName, appFound)
 	replaceTag(tags, "service", serviceName, svcFound)
 
+	redMetricsCustomTags := t.application.Map()
+	if t.redMetricsCustomTagKeys != nil {
+		for _, key := range t.redMetricsCustomTagKeys {
+			if value, found := getAppTag(key, "", span.Tags); found {
+				tags[key] = value
```
Comment (kentwang929) [inline reporter/reporter.go]: can redMetricsCustomTagKeys ever contains key that overlaps with previous tag keys?

File: reporter/reporter.go [hunk 2]
```diff
@@ -75,14 +76,22 @@ func LogPercent(percent float32) Option {
...
-		source:      hostname(),
-		application: app,
-		logPercent:  0.1,
-		bufferSize:  50000,
+		sender:                  sender,
+		source:                  hostname(),
+		application:             app,
+		logPercent:              0.1,
+		bufferSize:              50000,
+		redMetricsCustomTagKeys: nil,
```
Comment (kentwang929) [inline reporter/reporter.go]: doesn't seem necessary to nil it.  empty default will work.

File: reporter/reporter.go [hunk 3]
```diff
@@ -236,6 +245,17 @@ func (t *reporter) reportDerivedMetrics(span tracer.RawSpan) {
...
 
+	redMetricsCustomTags := t.application.Map()
+	if t.redMetricsCustomTagKeys != nil {
+		for _, key := range t.redMetricsCustomTagKeys {
+			if value, found := getAppTag(key, "", span.Tags); found {
+				tags[key] = value
+				redMetricsCustomTags[key] = value
+			}
+		}
+		t.heartbeater.Send(redMetricsCustomTags)
```
Comment (kentwang929) [inline reporter/reporter.go]: not sure the context for this.  heartbeat no longer functions as a heartbeat, but more of reporter that reports metrics called heartbeat?

File: reporter/reporter.go [hunk 4]
```diff
@@ -236,6 +244,21 @@ func (t *reporter) reportDerivedMetrics(span tracer.RawSpan) {
 	replaceTag(tags, "application", appName, appFound)
 	replaceTag(tags, "service", serviceName, svcFound)
 
+	redMetricsCustomTags := make(map[string]string)
```
Comment (vikramraman) [inline reporter/reporter.go]: Move this under the if condition? i.e create only if needed.

**Pull request merged**